crates/sui-indexer-alt/src/pipeline/concurrent/committer.rs

```diff
@@ -412,21 +264,14 @@ pub(super) fn committer<H: Handler + 'static>(
                     poll.reset_immediately();
                 }
 
-                Some(indexed) = rx.recv(), if pending_rows < H::MAX_PENDING_SIZE => {
+                Some(indexed) = indexed_rx.recv(), if pending_rows < H::MAX_PENDING_SIZE => {
                     metrics
                         .total_committer_rows_received
                         .with_label_values(&[H::NAME])
                         .inc_by(indexed.values.len() as u64);
 
-                    if indexed.values.is_empty() {
-                        // If the handler sends an empty batch, short-circuit the commit, and add
-                        // it directly to the pre-commit list.
-                        let (watermark, _) = indexed.into_batch();
-                        precommitted.insert(watermark);
-                    } else {
-                        pending_rows += indexed.values.len();
-                        pending.insert(indexed.cp_sequence_number, indexed);
-                    }
+                    pending_rows += indexed.values.len();
```

Comment:
We no longer get to take advantage of this shortcut here, but that's okay because we also simplified the guards to `poll.tick()` so it's no longer necessary.